x10/perpetual/contract.py

```diff
@@ -15,18 +17,41 @@
 STARK_PERPETUAL_ABI = "stark-perpetual.json"
 
 
-def call_stark_perpetual_withdraw(contract_address: str, eth_address: str, market: MarketModel, config: EndpointConfig):
+def call_stark_perpetual_withdraw(
+    contract_address: str,
+    eth_address: str,
+    market: MarketModel,
+    config: EndpointConfig,
+    get_eth_private_key: Callable[[], str],
+):
     web3_provider = Web3.HTTPProvider(config.chain_rpc_url, request_kwargs={"timeout": DEFAULT_API_TIMEOUT})
     web3 = Web3(web3_provider)
 
+    checksum_contract_address = Web3.to_checksum_address(contract_address)
+    checksum_eth_address = Web3.to_checksum_address(eth_address)
+
     contract = web3.eth.contract(
-        address=Web3.to_checksum_address(contract_address),
+        address=checksum_contract_address,
         abi=json.load(open(os.path.join(ABI_FOLDER, STARK_PERPETUAL_ABI), "r")),
     )
 
     method = contract.functions.withdraw(
-        int(eth_address, base=16),
+        int(checksum_eth_address, base=16),
         int(market.l2_config.collateral_id, base=16),
     )
 
-    return method.transact()
+    eth_private_key = get_eth_private_key()
+
+    assert is_hex_string(eth_private_key)
+
+    signed_tx = web3.eth.account.sign_transaction(
+        method.build_transaction(
+            {
+                "from": checksum_eth_address,
+                "nonce": web3.eth.get_transaction_count(checksum_eth_address),
+            }
+        ),
+        eth_private_key,
+    )
+
```

Comment:
It might make sense to `eth_private_key = None`, but need to do a memory dump and check when it will be collected by gc.

Reply:
lets see if we can harmonise between eth_account and web3 methods of signing / private key management - will do an investigation.

Reply:
hm, I lke 100% sure it is possible 'cause `web3` uses `eth_account` for this. but I think we should NOT "cache" or store  L1 private key in any long-lived variable inside SDK.